### PR TITLE
Basic obstacle avoidance

### PR DESCRIPTION
**Pull request by Quinstedt**
Simple Arduino sketch for basic obstacle avoidance assigment. The car will detect obstacles and stop, go backwards and continues in another direction with a small accelaration. The car will continue driving until it detects another obstacle.

new version of pull request #9

### CODE REVIEW

File: avoidance_car.ino [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+}
+
+void loop()
+{
+   
+    Serial.println(front.getDistance());
+    delay(100);
+    
+    if(front.getDistance() <= 100 && front.getDistance()>0){
```
Comment (platisd) [inline avoidance_car.ino:59]: What happens if the first call to `getDistance` returns `0` and the next one `150`? The car stops while it should not.

**Review by platisd — CHANGES_REQUESTED** [review]
Aside of the issue with the code, this sketch is not built by CI so there's no way to know if it actually compiles and nothing to keep someone from introducing a breaking change.

File: avoidance_car.ino (hunk 1 above)
Comment (Quinstedt) [reply avoidance_car.ino:59]: Hi, what do you mean with is not built by CI?
I just deleted the previous pull request (#9) and created a new one with the arduino file inside  ( pull request #10).

When distance is 0 then there is no obstacle detected and the car only stops if getDistance is bigger than 0 and less or equal to 100.

Comment (platisd) [reply avoidance_car.ino:59]: > Hi, what do you mean with is not built by CI?

It is not automatically compiled every time you push like the `smartcar.ino` does for example.

> When distance is 0 then there is no obstacle detected and the car only stops if getDistance is bigger than 0 and less or equal to 100.

Read the problem statement carefully:

> What happens if the first call to getDistance returns 0 and the next one 150?

Do you agree that the car will stop in that case?

**Pull request closed without merging**